### ISSUE TITLE
ClientQueryDuringMigrationsStressTest testQueryMapWithIndexes_whileShutdownStartup

### ISSUE THREAD

**Issue by jerrinot** [hazelcast/hazelcast#9724]
```
Error Message

[A query thread failed with: Not Member! target: [127.0.0.1]:5014, partitionId: -1, operation: com.hazelcast.map.impl.query.QueryOperation, service: hz:impl:mapService, A query thread failed with: Not Member! target: [127.0.0.1]:5014, partitionId: -1, operation: com.hazelcast.map.impl.query.QueryOperation, service: hz:impl:mapService]
Stacktrace

java.lang.AssertionError: [A query thread failed with: Not Member! target: [127.0.0.1]:5014, partitionId: -1, operation: com.hazelcast.map.impl.query.QueryOperation, service: hz:impl:mapService, A query thread failed with: Not Member! target: [127.0.0.1]:5014, partitionId: -1, operation: com.hazelcast.map.impl.query.QueryOperation, service: hz:impl:mapService]
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertFalse(Assert.java:64)
	at com.hazelcast.map.QueryDuringMigrationsStressTest$1.run(QueryDuringMigrationsStressTest.java:128)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueAllTheTime(HazelcastTestSupport.java:895)
	at com.hazelcast.map.QueryDuringMigrationsStressTest.queryDuringMigrations(QueryDuringMigrationsStressTest.java:125)
	at com.hazelcast.map.QueryDuringMigrationsStressTest.testQueryMapWithIndexes_whileShutdownStartup(QueryDuringMigrationsStressTest.java:116)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:88)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:82)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.lang.Thread.run(Thread.java:745)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.maintenance-nightly/com.hazelcast$hazelcast-client/1106/testReport/junit/com.hazelcast.client.map/ClientQueryDuringMigrationsStressTest/testQueryMapWithIndexes_whileShutdownStartup/

**Comment by jerrinot** [comment]
the test was [backported](https://github.com/jerrinot/hazelcast/commit/08393624fe1b750dbfb6f544e60bd10557bce021) into the maintenance branch, but the [original fix](https://github.com/hazelcast/hazelcast/pull/9098/files#diff-b8b92ca376e8d5beaa55a16b3b5c4852R111) was not.

**Comment by mdogan** [comment]
Should be fixed by https://github.com/hazelcast/hazelcast/pull/9725

**Comment by hasancelik** [comment]
```
java.lang.AssertionError: [Obtained 999 results for query 'id >= 20891 AND id < 21891', Obtained 996 results for query 'age >= 83770 AND age < 84770', Obtained 998 results for query 'id >= 96605 AND id < 97605', Obtained 998 results for query 'age >= 2996 AND age < 3996']
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertFalse(Assert.java:64)
	at com.hazelcast.map.QueryDuringMigrationsStressTest$1.run(QueryDuringMigrationsStressTest.java:128)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueAllTheTime(HazelcastTestSupport.java:951)
	at com.hazelcast.map.QueryDuringMigrationsStressTest.queryDuringMigrations(QueryDuringMigrationsStressTest.java:125)
	at com.hazelcast.map.QueryDuringMigrationsStressTest.testQueryMapWithIndexes_whileShutdownStartup(QueryDuringMigrationsStressTest.java:116)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:89)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:81)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.lang.Thread.run(Thread.java:745)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-nightly/com.hazelcast$hazelcast-client/1217/testReport/junit/com.hazelcast.client.map/ClientQueryDuringMigrationsStressTest/testQueryMapWithIndexes_whileShutdownStartup/

**Comment by vbekiaris** [comment]
This is fixed in 3.8. Some changes in `MapQueryEngineImpl` and the abstract client query task have been backported, but query engine has been refactored heavily so it's hard to locate missing parts of the solution, so closing.
